src/main/java/com/exactpro/th2/FixHandler.java

```diff
@@ -301,6 +302,13 @@ public CompletableFuture<MessageID> send(@NotNull ByteBuf body, @NotNull Map<Str
             throw new IllegalStateException("Session is not active. It is not possible to send messages.");
         }
 
+        FixField msgType = findField(body, MSG_TYPE_TAG);
+        boolean isLogout = msgType != null && Objects.equals(msgType.getValue(), MSG_TYPE_LOGOUT);
+        if(isLogout && !channel.isOpen()) {
+            LOGGER.warn("Logout ignored as channel is already closed.");
```

Comment:
Maybe create th2 event for both cases: gracefully and ungracefully logout by user request